### PR TITLE
[iOS] Fix sizing of button when using CharacterSpacing

### LINKED ISSUE #10293: Button does not resize to fit text with CharacterSpacing
### Description

Button is not sized to account for CharacterSpacing on iOS and MacCatalyst. 

iOS and MacCatalyst
Screen Shot 2022-09-23 at 11 06 55 AM

Android and Windows
Screen Shot 2022-09-23 at 12 40 41 PM

### Steps to Reproduce

1. Create a new .NET MAUI app
2. Modify the button on MainPage.xaml to include CharacterSpacing="1"
3. Run the application on iOS or MacCatalyst

Expected:
Screen Shot 2022-09-23 at 12 40 41 PM

Actual:
Screen Shot 2022-09-23 at 11 06 55 AM

### Link to public reproduction project repository

https://github.com/tdeborde2/MauiCharacterSpacing

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS 15.5, MacOS 12.6

### Did you find any workaround?

I was able to workaround the issue with a custom button control. 
This can be found in the repro project.

```cs
public class CustomButton : Button
{
	bool _sizeAllocated;

	double _oldCharacterSpacing;

	protected override void OnSizeAllocated(double width, double height)
	{
		if(!_sizeAllocated && WidthRequest == -1 && HorizontalOptions.Alignment != LayoutAlignment.Fill)
		{
			_sizeAllocated = true;
			this.WidthRequest = width + ((Text.Length - 1) * CharacterSpacing);
			return;
		}
		base.OnSizeAllocated(width, height);
	}

	protected override void OnPropertyChanging([CallerMemberName] string propertyName = null)
	{
		base.OnPropertyChanging(propertyName);
		if(propertyName == "Text")
		{
			if (_sizeAllocated)
			{
				WidthRequest = -1;
				_sizeAllocated = false;
			}
		}
	}

	protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
	{
		base.OnPropertyChanged(propertyName);
		if (_sizeAllocated && HorizontalOptions.Alignment != LayoutAlignment.Fill)
		{
			if (propertyName == "CharacterSpacing")
			{
				if (CharacterSpacing < _oldCharacterSpacing)
				{
					WidthRequest = Width - ((Text.Length - 1) * (_oldCharacterSpacing - CharacterSpacing));
				}
				if (CharacterSpacing > _oldCharacterSpacing)
				{
					WidthRequest = Width + ((Text.Length - 1) * (CharacterSpacing - _oldCharacterSpacing));
				}
			}
		}
		_oldCharacterSpacing = CharacterSpacing;
	}
}
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

We were setting the CharacterSpacing on the internal label while we can use the AttributedTitle on UIButton itself so it sizes correctly .

### Issues Fixed

Fixes #10293

### CODE REVIEW

**Review by mandel-macaque — APPROVED** [review]

**Comment by hartez** [comment]
@rmarinho   Looks like the device test needs to be updated. Maybe the `GetNativeCharacterSpacing()` method should be using `GetAttributedTitle()` instead of `TitleLabel`?

**Review by hartez — CHANGES_REQUESTED** [review]
Change looks good, need to bring the tests up to date.

**Review by hartez — APPROVED** [review]

File: src/Core/tests/DeviceTests/Handlers/Button/ButtonHandlerTests.iOS.cs [hunk 1]
```diff
@@ -86,7 +86,8 @@ Task PerformClick(IButton button)
 		{
 			var button = GetNativeButton(buttonHandler);
 
-			var attributedText = button.TitleLabel.AttributedText;
+			//var attributedText = button.TitleLabel.AttributedText;
```
Comment (jsuarezruiz) [inline src/Core/tests/DeviceTests/Handlers/Button/ButtonHandlerTests.iOS.cs:89]: Can we remove the commented line?

File: src/Core/tests/DeviceTests/Handlers/Button/ButtonHandlerTests.iOS.cs [hunk 2]
```diff
@@ -86,7 +86,8 @@ Task PerformClick(IButton button)
...
+			var attributedText = button.GetAttributedTitle(UIControlState.Normal);
```
Comment (jsuarezruiz) [inline src/Core/tests/DeviceTests/Handlers/Button/ButtonHandlerTests.iOS.cs:90]: The changes looks nice, and we already have tests validating the CharacterSpacing property but could we add a device test where we get the width of the native button before updating the CharacterSpacing and after setting it and check that the updated value is greater than the initial one?

**Review by jsuarezruiz — APPROVED** [review]

**Pull request merged**